pyclam/manifold.py

```diff
@@ -224,6 +223,9 @@ def radius(self) -> Radius:
         if 'radius' not in self.cache:
             logging.debug(f'building cache for {self}')
             _ = self.argradius
+            if self.cache['radius'] < 0:
+                raise ValueError(f'got cluster {self.name} with negative radius. '
+                                 f'Make sure that the distance function used always returns non-negative values.')
```

Comment:
I think this check belongs wherever we set `self.cache['radius']`

Reply:
moved